_posts/2020-01-01-SVGD.markdown

```diff
@@ -0,0 +1,405 @@
+    </p>
+  </details>
+  
+  2. "The open interval (0, 1) is not complete whereas the closed interval [0, 1] is complete." Why? Can we use this example to get a intuitive definition of complete?
+
+  <details><summary>Solution</summary>
+    <p>
+      Intuitively, a space is complete if there are no "points missing" from it (inside or at the boundary). For instance, the set of rational numbers is not complete, because e.g. \(\sqrt(2)\) is "missing" from it, even though one can construct a Cauchy sequence of rational numbers that converges to it. More information can be found at <a href="https://en.wikipedia.org/wiki/Complete_metric_space">Wikipedia: Complete Metric Space.</a>
+    </p>
+  </details>
+
+  3. Explain the difference between a Banach and Hilbert Space. Is every Hilbert space a Banach space?
+
+  <details><summary>Solution</summary>
+    <p>
+      A Banach space is a vector space in which each vector has a non-negative length, or norm, and in which every Cauchy sequence converges to a point of the space. Also known as complete normed linear space.
+
+      A Hilbert space is a Banach space with inner product, which defines the norm.
+    </p>
+  </details>
+
+  4. In Machine Learning, kernels can be thought of as a "dot product" (a kind of similarity score) in high-dimensional space. Why would this be useful? Given a feature map, do we always have a corresponding kernel? Given any kernel, can we always explicitly write out the elements of the corresponding feature map?
+
+  <details><summary>Solution</summary>
+    <p>
+      Kernels (and the corresponding kernel trick) allow us to compute similarities in high-dimensional space without explicitly writing out and computing the dot product. 
+
+      However, not ever feature map corresponds to a kernel; there are certain properties a kernel must have, and not every feature map imbues it with those properties.
+
+      Likewise, given a kernel, it may be the case that we can never write out (explicitly) the corresponding feature map. A good example of this is the popular exponential kernel.
+    </p>
+  </details>
+
+  5. What is the goal of supervised learning? In a high-dimensional, supervised setting, what makes this hard?
+
+  <details><summary>Solution</summary>
+    <p>
+      We want to learn a distribution \(p\) that fits the data generating distribution \(q\), while still being flexible enough to do learning and inference tractibly / efficiently. In high dimensions, the bias-variance tradeoff in model selection can hit us hard, along with another whole host of problems.
+    </p>
+  </details>
+
+  6. Assume that we just need the log-likeihood in many machine learning tasks so that we can compute  KL\((q||p)\) , and iteratively fit our model \(p\) to the underlying, generating data distribution \(q\). Why is this already too large of an assumption ("We assume that we have the ability to calculate the log-likelihood under the model that we specify")?
```

Comment:
In my screen, this paragraph shows up weird.

![Screen Shot 2020-01-20 at 7 29 29 PM](https://user-images.githubusercontent.com/615351/72765607-3e65bf00-3bbb-11ea-902e-f7d198cfe5e6.png)